### DISCUSSION TITLE
How to update dasel.New() usage for v2 API?

### DISCUSSION THREAD

**Discussion by beatcracker** [TomWright/dasel#376] [Q&A] [answered]
Previously there was a [Use as a go package](https://github.com/TomWright/dasel-docs/blob/v1/use-as-a-go-package.md) documentation page which also helped me to hack-in INI file support in my code using [zieckey/goini](https://github.com/zieckey/goini).

This relied on using `dasel.New(<map[string]string>)`, but for `v2` things changed significantly and this method doesn't exist anymore. Would love to get some help on this and maybe see a doc page make a return.

#### Example

```go
package foo

import (
	"github.com/tomwright/dasel"
	"github.com/zieckey/goini"
)

func modifyINI(path string, selector string, value string) (err error) {
	ini := goini.New()
	err = ini.ParseFile(path)
	if err != nil {
		return
	}

	kv := ini.GetAll()
	
	// This worked in V1
	node := dasel.New(kv)
	if err != nil {
		return
	}

	if value == nil {
		err = node.Delete(selector)
	} else {
		err = node.Put(selector, value)
	}

	if err != nil {
		return
	}

	// Write file...
	return
}
```

**Comment by TomWright** [comment, marked as answer]
In dasel V2 you can completely skip the step of creating a node and directly use package level functions `dasel.Select`, `dasel.Put` and `dasel.Delete`.

I have updated the docs: https://daseldocs.tomwright.me/examples/using-dasel-as-a-go-package

E.g.
```go
func modifyINI(path string, selector string, value string) (err error) {
	ini := goini.New()
	err = ini.ParseFile(path)
	if err != nil {
		return
	}

	kv := ini.GetAll()

        var result dasel.Value
	if value == nil {
		result, err = dasel.Delete(kv, selector)
	} else {
		result, err = dasel.Put(kv, selector, value)
	}

	if err != nil {
		return
	}

        resultValue := result.Interface()

	// Write file...

	return
}
```

**Reply by beatcracker** [reply]
Thanks, that really helped!

**Reply by TomWright** [reply]
If this ini package works well for you perhaps we could get it integrated into dasel itself. I'm pretty sure there's an ini request somewhere

**Reply by beatcracker** [reply]
The #50 , yes. To be honest I didn't test this thoroughly, can't vouch that it will cover all requirements.

**Reply by TomWright** [reply]
Hey @beatcracker, docs have been added. You don't actually have to use `dasel.ValueOf` and can instead pass your value in directly.

**Reply by beatcracker** [reply]
Awesome, thanks a lot! 🔥
